gcloud/bigtable/column_family.py

```diff
@@ -15,6 +15,59 @@
 """User friendly container for Google Cloud Bigtable Column Family."""
 
 
+class GarbageCollectionRule(object):
+    """Table garbage collection rule.
+
+    Cells in the table fitting the rule will be deleted during
+    garbage collection.
+
+    .. note::
+
+        At most one of ``max_num_versions`` and ``max_age`` can be specified
+        at once.
```

Comment:
Using separate classes seems much clearer to the user than requiring that she pass only one of N possible keyword args.  I can't tell from this usage (or the pattern in #1291) whether having a base class, `GarbageCollectionRule`, with two subclasses, `MaxAgeGCRule` and `MaxNumVersionsGCRule`, is better than two standalone classes.

What code marshals the rules to protobuf fields?